src/main/java/Task.java

```diff
@@ -0,0 +1,132 @@
+import java.io.IOException;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class Task {
+    protected String description;
+    protected boolean isDone;
+    protected boolean isCorrectFormat;
+
+    /**
+     * Constructor for Task.
+     *
+     * @param description Takes in a string that is
+     *     either Todo, Event or Deadline.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+        this.isCorrectFormat = false;
+    }
+
+    /**
+     * Method to get whether that date is in correct format.
+     *
+     * @return Returns the boolean isCorrectFormat for the date.
+     */
+    public boolean getIsCorrectFormat() {
+        return isCorrectFormat;
+    }
+
+    /**
+     * Returns tick or cross symbol.
+     *
+     * @return A tick or cross to symbolize whether the task has been done.
+     */
+    public String getStatusIcon() {
```

Comment:
Can add [] to the icons here, this would reduce complication and repetition of code